storage/persistence_test.go

```diff
@@ -0,0 +1,32 @@
+package storage
+
+import (
+	"context"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/evergreen-ci/logkeeper/model"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUploadBuildMetadata(t *testing.T) {
+	storage := makeTestStorage(t, "")
+	defer os.RemoveAll(tempDir)
+
+	build := model.Build{
+		Id:       "5a75f537726934e4b62833ab6d5dca41",
+		Builder:  "builder0",
+		BuildNum: 1,
+		Info:     model.BuildInfo{TaskID: "t0"},
+	}
+
+	assert.NoError(t, storage.UploadBuildMetadata(context.Background(), build))
+	results, err := storage.Get(context.Background(), "5a75f537726934e4b62833ab6d5dca41/metadata.json")
+	assert.NoError(t, err)
+	contents, err := io.ReadAll(results)
+	assert.NoError(t, err)
+
+	expectedMetadata := `{"id":"5a75f537726934e4b62833ab6d5dca41","builder":"builder0","buildnum":1,"task_id":"t0"}`
+	assert.Equal(t, expectedMetadata, string(contents))
```

Comment:
Just to confirm Go Json serialization always produces a stable key order?

Reply:
I think so ([docs](https://pkg.go.dev/encoding/json#:~:text=The%20map%20keys%20are%20sorted))